tests/catalog/manageregistration.go

```diff
@@ -0,0 +1,50 @@
+/*
+Copyright © 2022 SUSE LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package catalog
+
+import (
+	"github.com/rancher/elemental-operator/pkg/config"
+)
+
+type MachineRegistrationSpec struct {
+	MachineName                 string            `json:"machineName" yaml:"machineName"`
+	MachineInventoryLabels      map[string]string `json:"machineInventoryLabels" yaml:"machineInventoryLabels"`
+	MachineInventoryAnnotations map[string]string `json:"machineInventoryAnnotations" yaml:"machineInventoryAnnotations"`
+	Config                      *config.Config    `json:"config" yaml:"config"`
+}
+
+type MachineRegistration struct {
+	APIVersion string `json:"apiVersion" yaml:"apiVersion"`
+	Kind       string `json:"kind" yaml:"kind"`
+	Metadata   struct {
+		Name string `json:"name" yaml:"name"`
+	} `json:"metadata" yaml:"metadata"`
+	Spec MachineRegistrationSpec
+}
+
+func NewMachineRegistration(name string, config *config.Config) *MachineRegistration {
```

Comment:
What is the advantage of having a custom structure instead of importing machine registration from API package directly?